src/bap_mc/bap_mc.ml

```diff
@@ -170,6 +199,20 @@ module Cmdline = struct
     Arg.(value & opt_all ~vopt:`bil (enum formats) [] &
          info ["show-bil"] ~doc)
 
+  let addr =
+    let doc = "Specify an address of first byte, as though \
+	       the instructions occur at a certain address, \
+	       and accordingly interpreted. " in
+    Arg.(value & opt ~vopt:"0" (string) "0" &
+	 info ["addr"] ~doc)    
+
+  let max_insns =
+    let doc = "Specify a number of instructions to disassemble.\
+	       Good for ensuring that only one instruction is ever\
+	       lifted or disassembled from a byte blob. Default is all" in
+    Arg.(value & opt ~vopt:(Some 0) (some int) (Some 0) &
```

Comment:
No, you shouldn't ask users to input `--max-insn=0` for the infinity. The interface is as follows:
1. if `--max-insns` option is not specfied that means that there is no upper bound
2. otherwise, use the provided value as an upper-bound.

E.g.,
```
$ echo "DEADBEEFC0DE" | bap-mc 
```
will decode all instructions, but, 
```
$ echo "DEADBEEFC0DE" | bap-mc --max-insns=1
```
will stop after the first one.